packages/app-desktop/gui/JoplinCloudLoginScreen.tsx

```diff
@@ -0,0 +1,121 @@
+import { CSSProperties, useEffect, useReducer, useState } from 'react';
+import ButtonBar from './ConfigScreen/ButtonBar';
+import { _ } from '@joplin/lib/locale';
+import { AppState } from '../app.reducer';
+import { clipboard } from 'electron';
+import Button, { ButtonLevel } from './Button/Button';
+const bridge = require('@electron/remote').require('./bridge').default;
+import { uuidgen } from '@joplin/lib/uuid';
+import { Dispatch } from 'redux';
+import { reducer, intitialValues, generateLoginWithUniqueLoginCode, checkIfLoginWasSuccessful } from '@joplin/lib/services/JoplinCloudLogin';
+import Setting from '@joplin/lib/models/Setting';
+
+const { connect } = require('react-redux');
+const { themeStyle } = require('@joplin/lib/theme');
+
+const loginUrl = `${Setting.value('sync.10.website')}/login`;
+const applicationsUrl = `${Setting.value('sync.10.path')}/api/applications`;
```

Comment:
We shouldn't access settings in here because they may not be initialized when this file is instantiated. Instead you can get them from the state as you've done for the theme. So `state.settings['sync.10.website']`